src/components/Tooltip/Tooltip.tsx

```diff
@@ -56,113 +59,268 @@ const paddingsBottom = {
+                ]);
+            case currentPlacement.toString().includes(TooltipPosition.Left):
+                return merge([
+                    "before:tw-border-b-0 before:tw-border-l-0 tw-right-[-6px]",
+                    brightHeader && alignment === TooltipAlignment.Start
+                        ? brightHeaderArrowBackgroundColors[brightHeader]
+                        : "before:tw-bg-black-100 before:dark:tw-bg-white",
+                ]);
+            default:
+                return "before:tw-border-b-0 before:tw-border-r-0 tw-top-[-6px] before:tw-bg-black-100 before:dark:tw-bg-white";
+        }
+    };
+
+    const tooltipOffset = withArrow ? 10 : 5;
+    const popperInstance = usePopper(triggerRefElement?.current, tooltipContainerRef.current, {
+        placement: placement,
+        modifiers: [
+            {
+                name: "arrow",
+                options: {
+                    element: arrowElement,
+                },
+            },
+            {
+                name: "offset",
+                options: {
+                    offset: [0, tooltipOffset],
+                },
+            },
+            {
+                name: "flip",
+                enabled: flip,
+            },
+        ],
+    });
+
+    const arrowStyling = setArrowClasses();
+    const [isOpen, setIsOpen] = useState(false);
+    const timeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+    const handleShowTooltipOnHover = () => {
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current);
+        }
+        setIsOpen(true);
+    };
+
+    const handleHideTooltipOnHover = () => {
+        timeoutRef.current = setTimeout(() => setIsOpen(false), hoverDelay);
+    };
+
+    const checkIfHovered = (e: Event) => {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        const hoveredElement = e.path;
+        if (hoveredElement.includes(triggerRefElement?.current)) {
+            handleShowTooltipOnHover();
+        }
 
-    return (
-        <>
-            <div
-                ref={tooltipContainerRef}
-                className="tw-popper-container tw-inline-block tw-max-w-[200px] tw-bg-black-100 dark:tw-bg-white tw-rounded-md tw-shadow-mid tw-text-white dark:tw-text-black-100 tw-z-20"
-                style={style}
-                data-test-id="tooltip"
-                {...popperAttributes}
-                {...tooltipProps}
-            >
-                {brightHeader && <BrightHeader headerStyle={brightHeader} />}
+        if (hoveredElement.includes(tooltipContainerRef?.current)) {
+            handleShowTooltipOnHover();
+        }
+    };
+
+    const hasInteractiveElements = !!(buttons?.length || linkUrl?.length);
+    const firstButtonRef = useRef<HTMLButtonElement | null>(null);
+    const lastButtonRef = useRef<HTMLButtonElement | null>(null);
+
+    useEffect(() => {
```

Comment:
Better to add the events as props of `TooltipIcon` and use the `on...` notation in the component instead of vanilla events and ref